### PR TITLE
CDAP-1592: delete only aggregated metrics on removal of program, ...

### PR DESCRIPTION
**Pull request by abaranau**

### CODE REVIEW

**Comment by abaranau** [comment]
http://builds.cask.co/browse/CDAP-RBT188-1

**Comment by gsps1** [comment]
so we don't want to change MetricDeleteQuery as you mentioned in Jira ?

**Comment by abaranau** [comment]
you mean remove start/stop ts?

**Comment by abaranau** [comment]
we keep it, yes, for now - would make to many changes it seems

File: cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricStore.java [hunk 1]
```diff
@@ -220,8 +226,18 @@ public void delete(MetricDeleteQuery query) throws Exception {
...
+    // this will delete all aggregates metrics data
+    delete(new MetricDeleteQuery(0, System.currentTimeMillis() / 1000, null,
+                                             Maps.<String, String>newHashMap()));
+    // this will delete all timeseries data
+    deleteBefore(System.currentTimeMillis() / 1000);
+  }
+
   private CubeDeleteQuery buildCubeDeleteQuery(MetricDeleteQuery query) {
-    return new CubeDeleteQuery(query.getStartTs(), query.getEndTs(),
+    // note: delete query currently usually executed in sync way, so we only attempt to delete totals, to avoid timeout
```
Comment (gsps1) [inline cdap-watchdog/src/main/java/co/cask/cdap/metrics/store/DefaultMetricStore.java]: not executed asynchronously - might be better to understand.

File: cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsSuiteTestBase.java [hunk 2]
```diff
@@ -128,6 +131,9 @@ public static void beforeClass() throws Exception {
     StoreFactory storeFactory = injector.getInstance(StoreFactory.class);
     store = storeFactory.create();
     locationFactory = injector.getInstance(LocationFactory.class);
+
+    metricStore = injector.getInstance(MetricStore.class);
+
```
Comment (gsps1) [inline cdap-gateway/src/test/java/co/cask/cdap/gateway/handlers/metrics/MetricsSuiteTestBase.java]: remove unwanted new lines

File: cdap-watchdog-api/src/main/java/co/cask/cdap/api/metrics/MetricStore.java [hunk 3]
```diff
@@ -51,6 +51,12 @@
   void delete(MetricDeleteQuery query) throws Exception;
 
   /**
+   * Deletes all metrics data.
+   * @throws Exception
```
Comment (gsps1) [inline cdap-watchdog-api/src/main/java/co/cask/cdap/api/metrics/MetricStore.java]: does it make sense to add a note ,should only  be used in tests ?

**Comment by abaranau** [comment]
http://builds.cask.co/browse/CDAP-RBT188-3 GREEN

**Pull request merged**